lib/packwerk/reference_checking/checkers/privacy_checker.rb

```diff
@@ -23,10 +23,11 @@ def invalid_reference?(reference)
           return false if reference.constant.public?
 
           privacy_option = reference.constant.package.enforce_privacy
-          return false if enforcement_disabled?(privacy_option)
-
+          enforcement_disabled = privacy_option == false || privacy_option.nil?
```

Comment:
Why is this reimplementing `enforcement_disabled?`?

Reply:
This allowed the type to be refined through flow sensitivity. At any rate I reverted this after no longer declaring `Reference` as a `T::Struct`.